common/autoware_overlay_rviz_plugin/autoware_minimap_overlay_rviz_plugin/src/minimap.cpp

```diff
@@ -0,0 +1,614 @@
+    SLOT(updateLatitude()));
+
+  property_origin_lon_ = new rviz_common::properties::FloatProperty(
+    "Map Origin Longitude", 139.9009591876285, "Longitude of the map origin position", this,
+    SLOT(updateLongitude()));
+
+  zoom_ = property_zoom_->getInt();
+
+  latitude_ = property_latitude_->getFloat();
+  longitude_ = property_longitude_->getFloat();
+
+  tile_field_ = std::make_unique<TileField>(this);
+  tile_field_->setUrlTemplate(
+    TileProvider::getUrlTemplate(current_tile_provider_));  // Set the URL template
+
+  connect(tile_field_.get(), &TileField::tilesUpdated, this, &VehicleMapDisplay::onTilesUpdated);
+}
+
+VehicleMapDisplay::~VehicleMapDisplay()
+{
+}
+
+void VehicleMapDisplay::onInitialize()
+{
+  rviz_rendering::RenderSystem::get()->prepareOverlays(scene_manager_);
+  static int count = 0;
+  std::stringstream ss;
+  ss << "AerialMapDisplayObject" << count++;
+  overlay_.reset(new autoware_minimap_overlay_rviz_plugin::OverlayObject(ss.str()));
+  overlay_->show();
+  updateOverlaySize();
+  updateOverlayPosition();
+
+  node_ = context_->getRosNodeAbstraction().lock()->get_raw_node();
+
+  // Initialize the goal pose subscriber
+  goal_pose_sub_ = node_->create_subscription<geometry_msgs::msg::PoseStamped>(
+    "/planning/mission_planning/echo_back_goal_pose", 10,
+    std::bind(&VehicleMapDisplay::goalPoseCallback, this, std::placeholders::_1));
+
+  // pose_sub_ = node_->create_subscription<nav_msgs::msg::Odometry>(
+  //   "/localization/kinematic_state", 10,
+  //   std::bind(&VehicleMapDisplay::poseCallback, this, std::placeholders::_1));
+
+  pose_sub_ = node_->create_subscription<autoware_adapi_v1_msgs::msg::VehicleKinematics>(
+    "/api/vehicle/kinematics", rclcpp::QoS(10).best_effort(),
+    std::bind(&VehicleMapDisplay::poseCallback, this, std::placeholders::_1));
+
+  route_state_sub_ = node_->create_subscription<autoware_adapi_v1_msgs::msg::RouteState>(
+    "/api/routing/state", 10,
+    std::bind(&VehicleMapDisplay::routeStateCallback, this, std::placeholders::_1));
+
+  route_points_sub_ = node_->create_subscription<autoware_planning_msgs::msg::Trajectory>(
+    "/planning/scenario_planning/trajectory", 10,
+    std::bind(&VehicleMapDisplay::routePointsCallback, this, std::placeholders::_1));
+
+  // Subscribe to the map projector info topic
+  rclcpp::QoS qos_settings = rclcpp::QoS(1).reliable().transient_local();
+
+  map_projector_info_sub_ = node_->create_subscription<tier4_map_msgs::msg::MapProjectorInfo>(
+    "/map/map_projector_info", qos_settings,
+    std::bind(&VehicleMapDisplay::mapProjectorInfoCallback, this, std::placeholders::_1));
+
+  // Timer for smooth update
+  node_->create_wall_timer(
+    std::chrono::milliseconds(16), std::bind(&VehicleMapDisplay::smoothUpdate, this));
+}
+
+void VehicleMapDisplay::mapProjectorInfoCallback(
+  const tier4_map_msgs::msg::MapProjectorInfo::SharedPtr msg)
+{
+  map_projector_info_msg_ = msg;
+
+  projector_type_ = msg->projector_type;
```

Comment:
If we only support MGRS and UTM, we might want to send an error message.

```suggestion
  if (msg->projector_type != tier4_map_msgs::msg::MapProjectorInfo::MGRS && msg->projector_type != tier4_map_msgs::msg::MapProjectorInfo::LOCAL_CARTESIAN_UTM )
  {
    RCLCPP_ERROR_STREAM(
      node_->get_logger(),
      "Minimap plugin only supports MGRS and UTM for projector type"
  }
  projector_type_ = msg->projector_type;
```